src/lib/jws/jwsSigner.js

```diff
@@ -95,7 +97,12 @@ class JwsSigner {
 
         requestOptions.headers['fspiop-signature'] = JSON.stringify(signatureObject);
 
-        requestOptions.body = JSON.stringify(requestOptions.body);
+        if(typeof(requestOptions.body) !== 'string') {
+            requestOptions.body = JSON.stringify(requestOptions.body);
+        }
+        if(typeof(requestOptions.data) !== 'string') {
+            requestOptions.data = JSON.stringify(requestOptions.data);
+        }
```

Comment:
I think this can be simplified as
```
if(typeof(payload) !== 'string') {
   payload = JSON.stringify(payload);
}
requestOptions.data = requestOptions.body = payload;
```